aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/DeleteHandler.java

```diff
@@ -1,27 +1,96 @@
+import software.amazon.ssm.maintenancewindow.translator.ExceptionTranslator;
+import software.amazon.ssm.maintenancewindow.util.ClientBuilder;
+
+import java.util.Optional;
 
 public class DeleteHandler extends BaseHandler<CallbackContext> {
 
+    private static final SsmClient SSM_CLIENT = ClientBuilder.getClient();
+    private final ExceptionTranslator exceptionTranslator;
+
+    DeleteHandler() {
+        this.exceptionTranslator = new ExceptionTranslator();
+    }
+
+    /**
+     * Used for unit tests.
+     *
+     * @param exceptionTranslator Used for translating service model exceptions.
+     */
+    DeleteHandler(final ExceptionTranslator exceptionTranslator) {
+        this.exceptionTranslator = exceptionTranslator;
+    }
+
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        logger.log(String.format("Processing DeleteHandler request %s", request));
 
         final ResourceModel model = request.getDesiredResourceState();
 
-        // TODO : put your code here
+        final ProgressEvent<ResourceModel, CallbackContext> progressEvent =
+                ProgressEvent.<ResourceModel, CallbackContext>builder()
+                        .resourceModel(model)
+                        .build();
+
+        progressEvent.setStatus(OperationStatus.FAILED);
+
+        final Optional<DeleteMaintenanceWindowRequest.Builder> optionalRequestBuilder = initializeRequestBuilder(model);
+
+        if (!optionalRequestBuilder.isPresent()) {
+            progressEvent.setErrorCode(HandlerErrorCode.InvalidRequest);
+            progressEvent.setMessage("WindowId must be specified to delete a maintenance window.");
+            return progressEvent;
+        }
+
+        final DeleteMaintenanceWindowRequest deleteMaintenanceWindowRequest = optionalRequestBuilder.get().build();
+
+        try {
+            proxy.injectCredentialsAndInvokeV2(deleteMaintenanceWindowRequest, SSM_CLIENT::deleteMaintenanceWindow);
+            progressEvent.setStatus(OperationStatus.SUCCESS);
+        } catch (final Exception e) {
+            final BaseHandlerException cfnException = exceptionTranslator
+                    .translateFromServiceException(e, deleteMaintenanceWindowRequest);
+
+            logger.log(cfnException.getCause().getMessage());
+
+            throw cfnException;
+        }
+
+        if (progressEvent.isSuccess()) {
+            // nullify the model if delete succeeded
+            progressEvent.setResourceModel(null);
+        }
+
+
+        return progressEvent;
```

Comment:
Nit-pick: could create the success PE object here all at once, instead of mutating it to Success on-the-go. For example, `return ProgressEvent.defaultSuccessHandler(null);`

Reply:
Addressed